Sources/SyntaxKit/Theme.swift

```diff
@@ -0,0 +1,198 @@
+//
+//  Theme.swift
+//  SyntaxKit
+//
+//  Represents a TextMate theme file (.tmTheme). Currently only supports the
+//  foreground text color attribute on a local scope.
+//
+//  Created by Sam Soffes on 10/11/14.
+//  Copyright © 2014-2015 Sam Soffes. All rights reserved.
+//
+
+#if os(macOS)
+    import AppKit
+#else
+    import UIKit
+#endif
+
+public struct Theme {
+    
+    // MARK: - Public Types
+    
+    public typealias FontCallback = (_ fontName: String, _ fontSize: CGFloat, _ fontStyle: FontStyle) -> (Font?)
+    
+    // MARK: - Private Defaults
+    
+    private var fontName: String = Theme.defaultFontName
+    private var fontSize: CGFloat = Theme.defaultFontSize
+    private var fontCallback: FontCallback?
```

Comment:
The theme has nothing to do with font size or font name which should be selected by the application based on other criteria. So the FontCallback should return a font for a given FontStyle which it then uses. Alternatively, instead of a callback a struct with one font for each style could be passed in.

Reply:
Yes. But I am not sure if theme will control font family and font size:  https://github.com/textmate/textmate/blob/e4e54440b509557b79b2bd6f7b54bca279a88358/Frameworks/theme/src/theme.cc#L59

Reply:
Even if it is supported by textmate, I think these properties should simply be ignored. The text editor should always control typeface and font size (especially since installed fonts are something that differ from machine to machine, worse the same font might be named differently). I don't know any text editor which only lets the theme control these two properties (either it is fixed, or the user can choose).